src/main/scala/com/framex/core/FrameX.scala

```diff
@@ -19,9 +19,9 @@ class FrameX(var data: Vector[Vector[ElemX]], var columnMap: Map[String, Int] =
     (data(0).size, data.size)
   }
 
-  def head(n: Int = 5) : FrameX  = FrameX(data.map(c => c.slice(0, n)))
+  def head(n: Int = 5) : FrameX  = FrameX(data.map(c => c.slice(0, n)), this.columnNames)
 
-  def tail(n: Int = 5) : FrameX = FrameX(data.map(c => c.slice(c.size - n, c.size)))
+  def tail(n: Int = 5) : FrameX = FrameX(data.map(c => c.slice(c.size - n, c.size)), this.columnNames)
```

Comment:
If `n = 1`, does the result still have all the `columnNames`?

Reply:
```
FrameX(ll, columnNames).tail(1) shouldBe FrameX(
      Vector(
        Vector(ElemX(5)),
        Vector(ElemX("E")),
        Vector(ElemX("2017-02-12"))
      ),
      columnNames
    )
```
it sounds ok here. What's the bug...